Auth-MSSC/src/main/java/com/clinics/auth/model/User.java

```diff
@@ -0,0 +1,94 @@
+package com.clinics.auth.model;
+
+import com.clinics.auth.exception.validator.UniqueEmailConstraint;
+import com.clinics.common.security.Role;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.*;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor(access = AccessLevel.PUBLIC)
+@Builder(toBuilder = true)
+@Entity(name = "auth_user")
+public class User implements Role, Serializable, UserDetails{
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@JsonIgnore
+	private Long id;
+
+	@Builder.Default
+	@Column(updatable = false, nullable = false)
+	private UUID uuid = UUID.randomUUID();
+
+	@Column(unique = true)
+	@UniqueEmailConstraint
+	@NotBlank(message = "email is mandatory")
+	@Size(min = 3, max = 200, message = "email length out of range")
```

Comment:
If we have an @Email constraint, do we still need @Size or @NotBlank as well ? Feels like @UniqueEmailConstraint should be enough, too many annotations makes the code unreadable

Reply:
UniqueEmailconstraint is a different story, notBlank check if is null and size check size if I remember correctly